### PR TITLE
Added support for ephemeral state

### LINKED ISSUE #71: Implement ephemeral storage and update manifest schema
### Description

Implement the ephemeral storage feature to provide access to transient state associated directly with a participant, similar to the persistent state but with an additional parameter for the participant address. Modify the manifest schema to include the new event schema and handle changes such as classes being prefixed with `class`, e.g., `class.Master`.

### PR DESCRIPTION
**Pull request by sarvalabs-gokul**
### Description

This pull request implements the ephemeral storage feature to provide access to transient state associated directly with a participant, similar to the persistent state but with an additional parameter for the participant address. Additionally, this PR modifies the manifest schema to include the new event schema and handles changes such as classes being prefixed with `class`, e.g., `class.Master`.

- Closes #71

### CODE REVIEW

File: packages/js-moi-logic/src.ts/state/ephemeral-state.ts [hunk 1]
```diff
@@ -1,23 +1,79 @@
-import { ErrorCode, ErrorUtils } from "js-moi-utils";
+import { ErrorCode, ErrorUtils, hexToBytes } from "js-moi-utils";
+import { EntityBuilder } from "./entity-builder";
+import { AccessorBuilder, SlotAccessorBuilder } from "./accessor-builder";
+import { AbstractProvider } from "js-moi-providers";
+import { LogicDriver } from "../logic-driver";
+import { Schema, isPrimitiveType } from "js-moi-manifest";
+import { Depolorizer } from "js-polo";
+import { generateStorageKey } from "./accessor";
+
```
Comment (sarvalabs-harshrastogi) [inline packages/js-moi-logic/src.ts/state/ephemeral-state.ts:9]: To import tokens such as **`AccessorBuilder`**, **`LogicDriver`**, and **`AccessorBuilder`**, use **`import type`** or **`import { type NameOfImport } from "./package`**. If you use import type, it explicitly signals to the bundler that you don’t need the imported values, making tree shaking more effective.

**Review by sarvalabs-harshrastogi — CHANGES_REQUESTED** [review]

File: packages/js-moi-logic/src.ts/state/ephemeral-state.ts (hunk 1 above)
Comment (sarvalabs-gokul) [reply packages/js-moi-logic/src.ts/state/ephemeral-state.ts]: [7e514ee](https://github.com/sarvalabs/js-moi-sdk/pull/70/commits/7e514ee571ba04dffbb2c5224f0dcd48c6dee2e4)

File: packages/js-moi-logic/src.ts/logic-driver.ts [hunk 2]
```diff
@@ -100,16 +104,7 @@ export class LogicDriver<T extends Record<string, (...args: any) => any> = any>
...
-    }
-
-    /**
-     * Returns the interaction type for the logic driver.
-     * 
-     * @returns {IxType} The interaction type.
-     */
-    protected getIxType(): IxType {
-        return IxType.LOGIC_INVOKE;
+        return routine.mode === "persistent" || routine.mode === "ephemeral";
```
[the remark this answers is not in the document]
Comment (sarvalabs-gokul) [reply packages/js-moi-logic/src.ts/logic-driver.ts]: [7e514ee](https://github.com/sarvalabs/js-moi-sdk/pull/70/commits/7e514ee571ba04dffbb2c5224f0dcd48c6dee2e4)

File: packages/js-moi-providers/src.ts/base-provider.ts [hunk 3]
```diff
@@ -774,20 +772,57 @@ export class BaseProvider extends AbstractProvider {
...
+     * Retrieves the storage entry corresponding to a specific storage key, address and logic id.
+     * 
+     * @param {string} logicId - The logic id for which to retrieve the storage value.
+     * @param {string} storageKey - The storage key for which to retrieve the value.
+     * @param {string} address - The address related to the storage key (optional).
+     * @param {Options} options - The tesseract options. (optional)
+     * @returns {Promise<any>} A Promise that resolves to the storage value.
+     * @throws {Error} if there is an error executing the RPC call.
+     */
+    getStorageAt(logicId: string, storageKey: string, address: string, options?: Options): Promise<any>
```
[the remark this answers is not in the document]
Comment (sarvalabs-gokul) [reply packages/js-moi-providers/src.ts/base-provider.ts]: [7e514ee](https://github.com/sarvalabs/js-moi-sdk/pull/70/commits/7e514ee571ba04dffbb2c5224f0dcd48c6dee2e4)

**Pull request merged**